redbot/cogs/mod/settings.py

```diff
@@ -199,3 +199,17 @@ class ModSettings(MixinMeta):
                     command=f"{ctx.prefix}unban"
                 )
             )
+
+    @modset.command()
+    @commands.guild_only()
+    async def defaultdays(self, ctx: commands.Context, days: int = 0):
+        """Set the default days worth of message to be deleted when a user is banned.
+        The maximum amount of days is 7.
+        """
+        guild = ctx.guild
+        await self.settings.guild(guild).default_days.set(days)
```

Comment:
You have to check if days value is between 0 and 7 here

Reply:
Made it so if the number is lower than 0 it sets it to 0, if the number is larger than 7 it sets it to 7.